atomics/T1090.001/T1090.001.yaml

```diff
@@ -35,19 +35,17 @@ atomic_tests:
       default: "false"
   dependencies:
   - description: Squid must be installed and running
-    prereq_command: if [ -x "$(command -v squid --version)" ]; then exit 0; else exit 1; fi;
+    prereq_command: if ([ "#{run_squid}" = "false" ] || ([ -x "$(command -v squid --version)" ] && (ps aux | grep "squid -N" | grep -v "grep"))); then exit 0; else exit 1; fi;
     get_prereq_command: |-
-      if [[ "#{run_squid}" == "true" ]]; then sudo DEBIAN_FRONTEND=noninteractive apt-get -y install squid curl || sudo yum install -y squid curl || sudo dnf install -y squid curl || brew install curl squid; sed -i 's/http_access deny all/#http_access deny all/' /etc/squid/squid.conf; echo "running squid";squid -N -d 1 &; fi
-      if [[ "#{run_squid}" == "false" ]]; then echo "Skipping proxy setup"; else echo "Value #{run_squid} for variable "run_squid" is not valid"; fi
+      [ "#{run_squid}" = "true" ] && ([ -x "$(command -v squid --version)" ] || sudo DEBIAN_FRONTEND=noninteractive apt-get -y install squid curl || sudo yum install -y squid curl || sudo dnf install -y squid curl || brew install curl squid) && (squid -N -d 1 &) && exit 0
+      echo "Value #{run_squid} for variable "run_squid" is not valid" && exit 1
   executor:
     command: |
       export #{proxy_scheme}_proxy=#{proxy_server}:#{proxy_port}
-      curl http://#{test_domain}
+      curl #{test_url}
     cleanup_command: |
-      unset http_proxy
-      unset https_proxy
-      kill -9 $(ps -aux | grep "squid -N" | grep -v "color=auto" | awk '{print $2}')
-      apt-get -y remove squid
+      unset #{proxy_scheme}_proxy
+      [ "#{run_squid}" == "true" ] && (kill -9 $(ps aux | grep "squid -N" | grep -v "grep" | awk '{print $2}'))
```

Comment:
Kill only when run_quid is set to true. Also changed it to use `ps aux` instead of "ps -aux". In macOS, `ps -aux` searches for processes for user "x"